submit/sub_dataproc.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+"""
+submit job by dataproc
+Suggested and programmed by Zhili<zhilizheng@outlook.com>
+License: MIT
+"""
+
+import os
+import click
+import hailtop.batch as hb
+
+from analysis_runner import dataproc
+from cpg_utils.config import get_config
+
+from cpg_utils.hail_batch import (
+    authenticate_cloud_credentials_in_job,
+    copy_common_env,
+    remote_tmpdir,
+    output_path,
+)
+
+config = get_config()
+service_backend = hb.ServiceBackend(
+    billing_project=config["hail"]["billing_project"], remote_tmpdir=remote_tmpdir()
+)
+batch = hb.Batch(name="Dataproc batch", backend=service_backend)
+
+
+@click.command()
+@click.option("--script", help="script to run")
+@click.option("--jobname", help="jobname")
+@click.option("--time", default="2h", help="max time")
+@click.option("--nworker", default=2, help="number of workers")
+@click.option("--worker", default=None, help="type of worker: e.g. 'n1-highmem-8'")
+def submit(script, jobname, time, nworker, worker):
+    cluster = dataproc.setup_dataproc(
```

Comment:
Do you actually need to run this on Dataproc? Can you just use Query-on-Batch instead?

Reply:
Thanks Leo! I am thinking I will have to mount the data if I use batch. I am not familiar with the details, but it seems the efficiency of mounting large dataset is not promising (?)
Also, it is less error-prone (just for myself) to use dataproc than using batch, so I chose dataproc.

Reply:
What do you mean by mounting the data? Access to anything on GCS should work transparently, you just save a level of indirection.

Reply:
It seems I have to mount the data if I have to use plink in batch jobs, but it is irrelevant to the current analysis (as it won't require plink).

Reply:
It seems I have to revise the scripts substantially if I use Query-on-Batch instead of using Dataproc.   Is it possible to use dataproc for this analysis? I will have to learn how to code it using Batch. (It seems dataproc has better support for Hail than batch, so it is easier to use Hail in dataproc?)

Reply:
Why would you have to revise the scripts? You can just run the exact same query code directly, without going through Dataproc?

Reply:
Oh! I didn't know that! Could you please share a sample script?

Reply:
Hi @lgruen @illusional ! I tried to follow Anna's scripts (e.g., https://github.com/populationgenomics/tob-wgs/blob/pseudobulk-run/scripts/rv_expression_association/pseudobulk_association/get_promoter_variants.py) to invoke Hail batch by init_batch(), but I got the following errors (https://batch.hail.populationgenomics.org.au/batches/381149/jobs/1). 
For this script, I don't know how to fit it into the second example (https://github.com/populationgenomics/cpg-utils/blob/main/documentation/hail_batch.md). Could you please send me an example to follow? Thank you!!

Reply:
I think you're running on multiple branches. The code you linked is correct (on [`mt` branch](https://github.com/populationgenomics/sgs-somatic-mutation/blob/mt/deCODE_by_CHR.py)), but the code you ran, and the code in this PR is old (and what you ran) does not match.

This code you've linked is correct: https://github.com/populationgenomics/tob-wgs/blob/d5733d9646a992de52f1e3fba3d71c1221c3bce9/scripts/rv_expression_association/pseudobulk_association/get_promoter_variants.py#L38

Reply:
Thanks @illusional! I am a bit confused. I just tested the code with `init_batch()` in branch `mt`. I thought I should be able to test different scripts in different branches? I just copied it to branch `deCODE` and ran at test level, and the job failed again (https://batch.hail.populationgenomics.org.au/batches/381161/jobs/1). It seems to be a permission issue?